apps/web/pages/api/cancel.ts

```diff
@@ -175,21 +176,49 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     bookingToDelete.user.credentials.push(FAKE_DAILY_CREDENTIAL);
   }
 
-  const apiDeletes = async.mapLimit(bookingToDelete.user.credentials, 5, async (credential: Credential) => {
-    const bookingRefUid = bookingToDelete.references.filter((ref) => ref.type === credential.type)[0]?.uid;
-    const bookingExternalCalendarId = bookingToDelete.references.filter(
-      (ref) => ref.type === credential.type
-    )[0]?.externalCalendarId;
-    if (bookingRefUid) {
-      if (credential.type.endsWith("_calendar")) {
-        const calendar = getCalendar(credential);
-
-        return calendar?.deleteEvent(bookingRefUid, evt, bookingExternalCalendarId);
-      } else if (credential.type.endsWith("_video")) {
-        return deleteMeeting(credential, bookingRefUid);
+  const apiDeletes = [];
+
+  const calendarReference = bookingToDelete.references.filter((reference) =>
+    reference.type.includes("_calendar")
+  )[0];
+
+  if (calendarReference) {
+    const { credentialId, uid, externalCalendarId } = calendarReference;
+    if (credentialId) {
+      const calendarCredential = bookingToDelete.user.credentials.find(
+        (credential) => credential.id === credentialId
+      );
+
+      if (calendarCredential) {
+        const calendar = getCalendar(calendarCredential);
+        apiDeletes.push(calendar?.deleteEvent(uid, evt, externalCalendarId));
```

Comment:
Is there a better way to does this rather than nesting if statements?

Reply:
You can take out nested conditional, and his if condition can have both values.
```suggestion
  if (calendarReference) {
    const { credentialId, uid, externalCalendarId } = calendarReference;
    if (credentialId) {
      const calendarCredential = bookingToDelete.user.credentials.find(
        (credential) => credential.id === credentialId
      );
     }
      if (credentialId && calendarCredential) {
        const calendar = getCalendar(calendarCredential);
        apiDeletes.push(calendar?.deleteEvent(uid, evt, externalCalendarId));
      }

     
```

Reply:
Or extract this into a function and use early returns.